collect_app/src/main/java/org/odk/collect/android/activities/GeoPointMapActivity.java

```diff
@@ -259,6 +320,63 @@ public void initMap(MapFragment newMapFragment) {
 
         map.setGpsLocationListener(this::onLocationChanged);
         map.setGpsLocationEnabled(true);
+
+        if (instanceState != null) {
```

Comment:
initMap() method is called in onCreate(), could you explain why you decided to restore state here not just in onCreate (like we do in other cases) then you wouldn't need that additional instanceState field.

Reply:
Yeah, I think this approach is better than what I did in GeoTrace and GeoShape.  In those classes, it's not so easy to see that the restoring of state is done correctly, because it's intermingled with the rest of the setup in `initMap()`.  `initMap()` is already quite long and does a lot of things.  (Maybe it's not so bad with GeoShape because there are only 3 fields to restore, but GeoTrace and GeoPoint are more complicated.)

My thinking was, putting all of it in a separate method `restoreFromInstanceState()` guarantees that all of the restoring happens after `initMap()`, so that it doesn't get accidentally overwritten due to things happening in the wrong order.

Does that seem right to you?

Reply:
so, in this case, it's risky to read those values before but in GeoTraceActivity or GeoShapeActivity it's ok?

Reply:
Hmm, I'm not sure I understand your question.  I think the way that instance state is handled in GeoTraceActivity is harder to maintain than the way it's done in this PR.  If I'm failing to address your concern, sorry!—could you spell out the problem or risk you're concerned about?